src/common/http.hpp

```diff
@@ -125,7 +125,7 @@ string MethodToString(Method method);
 struct BrokenDownUrl {
 	string protocol;
 	string host;
-	int port {-1};
+	uint16_t port;
```

Comment:
If what you are concern here is the sign, isn't better to use `unsigned int` and let the compiler chose the most efficient size?

Reply:
No, it's not the signedness. The range of network port numbers fit exactly into the `uint16_t` type, which means we can use the json getter with range checking that I made above, and get the range checking for free. That would not work with a bigger type.

It's also the fact that when passing it to Boost it also uses `uint16_t`, so we don't need a cast there.

Reply:
Thanks for the clarification.